engine/consensus/matching/engine_test.go

```diff
@@ -57,421 +52,187 @@ func TestMatchingEngine(t *testing.T) {
-	// might forward us Execution Receipts from others for blocks they haven't computed themselves
-	ms.T().Skip()
-
-	// try to submit a receipt with a random origin ID
-	originID := ms.exeID
-	receipt := unittest.ExecutionReceiptFixture()
-
-	err := ms.matching.onReceipt(originID, receipt)
-	ms.Require().Error(err, "should reject receipt with mismatching origin and executor")
-
-	ms.resultsDB.AssertNumberOfCalls(ms.T(), "Store", 0)
-	ms.resultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
-	ms.sealsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
-}
-
-// try to submit a receipt from a NON-ExecutionNode  (here: consensus node)
+// Test that we reject receipts for unknown blocks without generating an error
 func (ms *MatchingSuite) TestOnReceiptUnknownBlock() {
-	originID := ms.conID
-	receipt := unittest.ExecutionReceiptFixture(unittest.WithExecutorID(originID))
+	// This receipt has a random block ID, so the matching engine won't find it.
+	receipt := unittest.ExecutionReceiptFixture()
 
-	// onReceipt should not throw an error
-	err := ms.matching.onReceipt(originID, receipt)
-	ms.Require().NoError(err, "should ignore receipt for unknown block")
+	// onReceipt should reject the receipt without throwing an error
+	err := ms.matching.onReceipt(receipt.ExecutorID, receipt)
+	ms.Require().NoError(err, "should drop receipt for unknown block without error")
 
-	ms.resultsDB.AssertNumberOfCalls(ms.T(), "Store", 0)
-	ms.resultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
-	ms.sealsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
+	ms.ReceiptsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
+	ms.ResultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
 }
 
-// try to submit a receipt for a known block from a consensus node
-func (ms *MatchingSuite) TestOnReceiptInvalidRole() {
-	originID := ms.conID
+// matching engine should drop Result for known block that is already sealed
+// without trying to store anything
+func (ms *MatchingSuite) TestOnReceiptSealedResult() {
+	originID := ms.ExeID
 	receipt := unittest.ExecutionReceiptFixture(
 		unittest.WithExecutorID(originID),
-		unittest.WithResult(unittest.ExecutionResultFixture(unittest.WithBlock(&ms.unfinalizedBlock))),
+		unittest.WithResult(unittest.ExecutionResultFixture(unittest.WithBlock(&ms.LatestSealedBlock))),
 	)
 
 	err := ms.matching.onReceipt(originID, receipt)
-	ms.Require().Error(err, "should reject receipt from wrong node role")
-	ms.Require().True(engine.IsInvalidInputError(err))
+	ms.Require().NoError(err, "should ignore receipt for sealed result")
 
-	ms.resultsDB.AssertNumberOfCalls(ms.T(), "Store", 0)
-	ms.resultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
-	ms.sealsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
+	ms.ResultsDB.AssertNumberOfCalls(ms.T(), "Store", 0)
+	ms.ResultsPL.AssertNumberOfCalls(ms.T(), "Add", 0)
 }
 
-// try ot submit a receipt from an Execution node, with zero unstaked node
-func (ms *MatchingSuite) TestOnReceiptUnstakedExecutor() {
```

Comment:
checking Receipt validity is now handled by `ReceiptValidator`. For the matching engine, we only need to test that the output of `ReceiptValidator` is processed properly: see test https://github.com/onflow/flow-go/blob/32081ff9f91d24fd5c10ab57bdf447a70c7dd4ef/engine/consensus/matching/engine_test.go#L212-L213